akka-actor/src/main/scala/akka/util/WildcardTree.scala

```diff
@@ -18,15 +18,24 @@ private[akka] final case class WildcardTree[T](data: Option[T] = None, children:
       copy(data = Some(d))
     } else {
       val e = elems.next()
-      copy(children = children.updated(e, children.get(e).getOrElse(WildcardTree()).insert(elems, d)))
+      copy(children = children.updated(e, children.getOrElse(e, WildcardTree()).insert(elems, d)))
     }
 
-  @tailrec final def find(elems: Iterator[String]): WildcardTree[T] =
-    if (!elems.hasNext) this
-    else {
-      (children.get(elems.next()) orElse children.get("*")) match {
-        case Some(branch) ⇒ branch.find(elems)
-        case None         ⇒ WildcardTree()
-      }
+  def find(elems: Iterator[String]): WildcardTree[T] = {
```

Comment:
The way we juggle the elems is amazingly inefficient. `ActorPath.elements` creates a `List` which we convert to an `Iterator` in the `lookup` and then we convert it back to a `List` here.
I think that can be improved by adding a `private[akka] def find(elems: List[String]): WildcardTree[T]` here and use that in `Deployer.lookup(ActorPath)`.

Perhaps we should also consider a `Map` cache for the `lookup` but that can be done in a separate PR.